STD-7 | Domain models (code)

zefirlover opened the pull request.

zefirlover committed:
Beta-version models added
Teacher was replaced, Course with its blocks added
Minor changes in config and DbContext

SleepyF0X commented on StudySharp.Domain/Models/Learning/Content.cs:
```diff
@@ -0,0 +1,16 @@
+        public int Id { get; set; }
+        public Course Course { get; set; }
+        
+        public List<TheoryBlock> TheoryBlocks { get; set; }
+        public TheoryBlock TheoryBlock { get; set; }
+        
+        public List<PracticalBlock> PracticalBlocks { get; set; }
+        public PracticalBlock PracticalBlock { get; set; }
```
For what do you use a Collection of models and a single model?

zefirlover replied:
Kola said that we need them both
I doubted, but add it like he said

zefirlover replied:
Models was remowed

SleepyF0X commented on StudySharp.Domain/Models/Learning/Course.cs:
```diff
@@ -0,0 +1,21 @@
+        public int Id { get; set; }
+        public string Name { get; set; }
+        public List<Tag> Tags { get; set; }
+        public DateTimeOffset CreatedAt { get; set; }
+        public DateTimeOffset RedactedAt { get; set; }
```
Its KolaCation idea?

zefirlover replied:
It WAS
Yes, it was

SleepyF0X replied:
Ok, set RedactedAt nullable

SleepyF0X commented on StudySharp.Domain/Models/Learning/Quiz.cs:
```diff
@@ -0,0 +1,8 @@
+    public class Quiz
+    {
+        public int Id { get; set; }
+        public PracticalBlock PracticalBlock { get; set; }
```
How do you get a PracticalBlock model without PracticalBlockId?

zefirlover replied:
Fixed

SleepyF0X commented on StudySharp.DomainServices/Configurations/TeacherConfiguration.cs:
```diff
@@ -12,7 +12,11 @@ public void Configure(EntityTypeBuilder<Teacher> builder)
+                .Property(_ => _.LastName)
                 .IsRequired();
         }
     }
```
Please, add similar configurations for another models

zefirlover committed:
New configs added
And minor changes in Quiz, Content and Tag

SleepyF0X commented on StudySharp.Domain/Models/Learning/Content.cs:
```diff
@@ -0,0 +1,14 @@
+    public class Content
+    {
+        public int Id { get; set; }
+        public Course Course { get; set; }
```
How do you get a Course model without CourseId?

@KolaCation we really need Id here?

zefirlover replied:
For the 1 to 1 connection we need it

SleepyF0X commented on StudySharp.DomainServices/Configurations/CourseConfiguration.cs:
```diff
@@ -0,0 +1,27 @@
+                .Property(_ => _.CreatedAt)
+                .IsRequired();
+
+            builder
+                .Property(_ => _.RedactedAt)
+                .IsRequired();
+        }
```
Not required field

zefirlover committed:
Minor changes
RedactedAt is nullable now
Content have new FK

zefirlover committed:
Nullable patch
RedactedAt in Course is nullable now

zefirlover committed:
Tag can have M-M connection to Course now
Also decorative //connection part added
We do not need it in Content, cause it consist of connection parts

zefirlover committed:
Removing encodings, indexlayout ang vcs from github
Were added in "Nullable patch" by mistake
